ocaml/tests/test_repository_helpers.ml

```diff
@@ -511,551 +511,1001 @@ module EvalGuidanceForOneUpdate = Generic.MakeStateless (struct
+      ; (* livepatch_guidance - None *)
+        ( ( ( [
+                ( "UPDATE-0000"
+                , UpdateInfo.
+                    {
+                      id= "UPDATE-0000"
+                    ; summary= "summary"
+                    ; description= "description"
+                    ; rec_guidance= Some Guidance.RebootHost
+                    ; abs_guidance= None
+                    ; guidance_applicabilities= [] (* No applicabilities *)
+                    ; spec_info= "special info"
+                    ; url= "https://update.details.info"
+                    ; update_type= "security"
+                    ; livepatch_guidance= None
+                    ; livepatches=
+                        [
+                          LivePatch.
+                            {
+                              component= Livepatch.Xen
+                            ; base_build_id=
+                                "9346194f2e98a228f5a595b13ecabd43a99fada0"
+                            ; base_version= "4.13.4"
+                            ; base_release= "10.22.xs8"
+                            ; to_version= "4.13.4"
+                            ; to_release= "10.23.xs8"
+                            }
+                          
+                        ]
+                    }
+                  
+                )
+              ]
+            , Update.
+                {
+                  name= "xen-hypervisor"
+                ; arch= "x86_64"
+                ; old_epoch= Some None
+                ; old_version= Some "4.13.4"
+                ; old_release= Some "10.22.xs8"
+                ; new_epoch= None
+                ; new_version= "4.13.4"
+                ; new_release= "10.23.xs8"
+                ; update_id= Some "UPDATE-0000"
+                ; repository= "regular"
+                }
+              
+            )
+          , ["UPDATE-0000"]
+          )
+        , None
+        )
+      ; (* livepatch_guidance: livepatch does not come from myself *)
+        ( ( ( [
+                ( "UPDATE-0000"
+                , UpdateInfo.
+                    {
+                      id= "UPDATE-0000"
+                    ; summary= "summary"
+                    ; description= "description"
+                    ; rec_guidance= None
+                    ; abs_guidance= None
+                    ; guidance_applicabilities= [] (* No applicabilities *)
+                    ; spec_info= "special info"
+                    ; url= "https://update.details.info"
+                    ; update_type= "security"
+                    ; livepatch_guidance= Some Guidance.RestartDeviceModel
+                    ; livepatches=
+                        [
+                          LivePatch.
+                            {
+                              component= Livepatch.Xen
+                            ; base_build_id=
+                                "9346194f2e98a228f5a595b13ecabd43a99fada0"
+                            ; base_version= "4.19.19"
```

Comment:
Is the point here that these versions don't match the ones in the Update below?

Reply:
Or that the update ids don't match? I don't understand "livepatch does not come from myself" in the comment above.

Reply:
I updated the comment to make it clearer. 
Here the `myself` refers to the one `update` in `eval_guidance_for_one_update`. This function evaluate guidance against this RPM update. But this evaluation needs to consider livepatch as well:
1. If  this RPM update has a livepatch (its update info is contained in the ones introduced by applicable livepatches), this is for the original comment "livepatch comes from myself"
2. Otherwise, "livepatch doesn't come from myself".
But anyway, I refined the comments and test data to make it clearer.